feat: support prettier-plugin-jsdoc

aminya opened the pull request:
Fixes #21

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dprint/dprint-plugin-prettier?pullRequest=27) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dprint/dprint-plugin-prettier?pullRequest=27) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dprint/dprint-plugin-prettier?pullRequest=27) it.</sub>

dsherret reviewed (changes requested):
Thanks @aminya and sorry for my delay. Would you be able to add a test for this in the ./tests/specs folder? That way we can ensure this works and keeps working. Thanks!

aminya commented:
I get an error that `exports` is not defined. I tried to fix it by resolving to ES6 source code. However, It doesn't fix it.

aminya commented:
I replaced EsBuild with Parceljs. Now, Parceljs correctly bundles the project making it possible to use it in the browser. 

I also added a test for JSDoc formatting.

dsherret reviewed (approved):
Thanks a lot for figuring this out, @aminya! Also, sorry for my delay.

I made a few changes to this PR in order to make the plugin opt-in since it doesn't seem that popular.

aminya commented:
Thanks for merging. 

I don't mind the opt-in option, but I think jsDoc is way more popular than Astro and Svelte. So the options are inconsistent.

dsherret commented:
I was thinking both astro and svelte would be opt-in by asking the dprint-plugin-prettier plugin to format svelte/astro files or not? In the case of jsdoc it's intertwined in the js output.

aminya commented:
Well, prettier-plugin-jsdoc also only works if the user uses TSDoc or JSDoc for documentation. The nature of the existence of a file format doesn't justify the feature. Or we could go the other way and enable all of them.

dsherret commented:
I’d rather this format TypeScript and JavaScript files out of the box the same way as prettier does, then people can opt in to the jsdoc plugin to change how ts/js files are formatted if they would like similar to how they have to opt in with prettier. Actually, maybe the option configuration should have the same structure as prettier does and be an array.
